src/lucky/verify_accepts_format.cr

```diff
@@ -0,0 +1,78 @@
+module Lucky::VerifyAcceptsFormat
+  abstract def clients_desired_format : Symbol
+
+  macro included
+    class_property _accepted_formats = [] of Symbol
+    before verify_accepted_format
+  end
+
+  macro accepted_formats(formats)
+    {% if !formats.is_a?(ArrayLiteral) %}
+      {% raise "#{@type} 'accepted_formats' should be an array of Symbols. Example: [:html, :json]" %}
+    {% end %}
+
+    {% if formats.size == 1 %}
+      accepted_formats {{ formats }}, default: {{ formats.first }}
+    {% else %}
+      {% formats.raise "#{@type} must pass a default to 'accepted_formats'. Example: accepted_formats [:html, :json], default: :html" %}
```

Comment:
I'm confused. If this gets raised when the `formats.size` isn't `1`, wouldn't that raise on your suggestion here which has two formats?

```crystal
accepted_formats [:html, :json], default: :html
```

Reply:
I think `formats` in this case is `[ [:html, :json], default: :html]`(ish)... So, if you had just `[:html, :json]` then it would fail because it's missing the `default`. However, if you passed `:html, :json` it wouldn't throw the error, but might still cause some issues. Maybe there's a better way to check this?

Reply:
There are 2 macros here. One has just one arg. That one can only accept a single element array. Anything else means you need to use the other macro below that sets an explicit default

Does that make sense?

Reply:
Ah ha! Yes I think I was thinking it would count the number of formats not the count of arguments passed. I.e. it's counting there is one argument, not one formats.

Reply:
It is counting that there is one format, but *only* if you are using the macro with one argument. If you are then you can only use 1 format. If  you use the one argument macro with multiple formats then we don't want to "guess" what the correct format is, we need to know the default. If you give it an empty array of formats then we also can't guess what the default tis so you need to tell Lucky explicitly.

If you look at the other macro below it does not check the number of formats. So you can use multiple or an empty array to allow all formats